### PR TITLE
feat: adds Proxy resource, and application_id on Reactor resource

### PR DESCRIPTION
**Pull request by brigonzalez**
## Description

- adds Proxy resource, and application_id on Reactor resource

## Testing required outside of automated testing?

- [x] Not Applicable

### Screenshots (if appropriate):

- [x] Not Applicable

## Rollback / Rollforward Procedure

- [x] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: internal/provider/resource_basistheory_proxy.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+		UpdateContext: resourceProxyUpdate,
+		DeleteContext: resourceProxyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "Unique identifier for the Proxy",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"key": {
```
Comment (koalaty-code) [inline internal/provider/resource_basistheory_proxy.go:29]: Should this be considered sensitive?

File: internal/provider/resource_basistheory_proxy.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+			"request_reactor_id": {
+				Description: "Request reactor ID for the Proxy",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"require_auth": {
+				Description: "Require auth for the Proxy",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
```
Comment (koalaty-code) [inline internal/provider/resource_basistheory_proxy.go:58]: This should default to `true` to match the behavior of the API, or we have to make it nullable to take the API's default value.

Comment (brigonzalez) [reply internal/provider/resource_basistheory_proxy.go:58]: when I tested this I got confused between the default bool in Go and the response from the API. good catch

File: internal/provider/resource_basistheory_proxy.go (hunk 1 above)
Comment (brigonzalez) [reply internal/provider/resource_basistheory_proxy.go:29]: I suppose we did move that from the URL to a header now 👍

**Review by koalaty-code — APPROVED** [review]

**Pull request merged**